llvm/include/llvm/Target/GlobalISel/Combine.td

```diff
@@ -1857,6 +1857,26 @@ class integer_of_opcode<Instruction castOpcode> : GICombineRule <
 
 def integer_of_truncate : integer_of_opcode<G_TRUNC>;
 
+def anyext_undef: GICombineRule<
+   (defs root:$root),
+   (match (G_IMPLICIT_DEF $undef),
+          (G_ANYEXT $root, $undef):$Aext),
+   (apply [{ Helper.replaceInstWithUndef(*${Aext}); }])>;
```

Comment:
` Helper.matchAnyExplicitUseIsUndef` checks in C++, if there is an undef on the anyext. It will fail 99.999999999999% of the time. Now it is a correct pattern and the combiner takes care of the matching.

I said in the summary `Is there a limit on the number of patterns?`.

Reply:
You can leave it where it is, and we can have another PR to convert `propagate_undef_any_op` into the new match syntax for *all* opcodes, not just undef. This PR is an optimization which is now mixing a refactoring. I'm going to approve this PR on the assumption that you leave undef combine where it is until we convert them en-mass.